src/main/java/hu/wurfel/refference/school/diary/DiaryController.java

```diff
@@ -1,40 +1,36 @@
 package hu.wurfel.refference.school.diary;
 
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.ArrayList;
 
-@RestController
+@RestController("/api/v1/Diary")
 public class DiaryController {
 
 
-    @Autowired
     DiaryService diaryService;
 
-    @PostMapping("/api/v1/search/Diary")
+    @PostMapping("/search")
     public ResponseEntity<ArrayList> searchForStudent(@RequestBody DiaryRequest diaryRequest) {
-        ArrayList answer = new ArrayList();
+        ArrayList answer;
         answer = diaryService.getAutomated(diaryRequest);
         return ResponseEntity.ok(answer);
     }
 
-	@PostMapping("/api/v1/adding/Diary")
+    @PostMapping("/adding")
     public ResponseEntity<ArrayList> adding(@RequestBody DiaryRequest diaryRequest) {
         ArrayList answer = new ArrayList();
         answer.add(diaryService.saveDiary(Integer.parseInt(diaryRequest.getId()), Long.parseLong(diaryRequest.getStudentId()), Integer.parseInt(diaryRequest.getClassId())));
         return ResponseEntity.ok(answer);
     }
 
-	@PostMapping("/api/v1/delete/Diary")
-    public void delete(@RequestBody DiaryRequest diaryRequest) {
-        diaryService.deleteDiary(diaryService.getDiaryByDiaryid(Integer.parseInt(diaryRequest.getId())));
+    @DeleteMapping("/delete/{id}")
+    public void deleteById(@PathVariable String id) {
+        diaryService.deleteDiary(diaryService.getDiaryByDiaryid(Integer.parseInt(id)));
```

Comment:
Itt is célszerű visszatérni ResponseEntity-vel. Gyakran van, hogy delete-re vissza szeretné kapni a frontend a törölt objektumot vagy visszaigazolást kér (ezt mondjuk mindig), hogy sikeres volt-e a művelet.